### PR TITLE
Dlg contrast

### PR DESCRIPTION
**Pull request by Lunalo**
It is now working okay but i have a question on how we can overwrite existing contrast columns using the checkbox ?

### CODE REVIEW

File: instat/dlgContrasts.vb [hunk 1]
```diff
@@ -46,57 +42,60 @@ Public Class dlgContrasts
...
-        ucrBase.clsRsyntax.iCallType = 2
         ucrReceiverForContrasts.Selector = ucrSelectorForContrast
         ucrReceiverForContrasts.SetMeAsReceiver()
         ucrReceiverForContrasts.SetIncludedDataTypes({"factor"})
         ucrBase.iHelpTopicID = 353
-        ucrFactorLevelsAndLabels.SetReceiver(ucrReceiverForContrasts)
-        ucrFactorLevelsAndLabels.SetAsSingleSelector()
-        clsLevels.SetRCommand("levels")
+        ucrInputContrast.SetItems({"Current Contrast", "Helmert", "Polynomials", "Treatment/Control", "Sum to Zero", "User Defined(Overwrite)"})
+        ucrBase.clsRsyntax.SetFunction(frmMain.clsRLink.strInstatDataObject & "$set_contrasts_of_factor")
```
Comment (dannyparsons) [inline instat/dlgContrasts.vb]: Why have you removed these parts?

**Comment by dannyparsons** [comment]
That's a good point, yes we should think about how to link these up at some point.

File: instat/dlgContrasts.vb (hunk 1 above)
Comment (Lunalo) [inline instat/dlgContrasts.vb]: I am not sure if we still need to run this using the base r function.

Comment (Lunalo) [inline instat/dlgContrasts.vb]: can we discuss this please

File: instat/dlgContrasts.vb [hunk 2]
```diff
@@ -95,7 +94,6 @@ Public Class dlgContrasts
                 ucrBase.clsRsyntax.AddParameter("new_contrasts", Chr(34) & "contr.sum" & Chr(34))
             Case "User Defined(Overwrite)"
```
Comment (dannyparsons) [inline instat/dlgContrasts.vb]: You can remove this case now.

File: instat/dlgContrasts.vb [hunk 3]
```diff
@@ -46,57 +41,59 @@ Public Class dlgContrasts
...
+        Select Case ucrInputContrast.GetText
+            Case "Treatment/Control"
+                ucrBase.clsRsyntax.AddParameter("new_contrasts", Chr(34) & "contr.treatment" & Chr(34))
+            Case "Helmert"
+                ucrBase.clsRsyntax.AddParameter("new_contrasts", Chr(34) & "contr.helmert" & Chr(34))
+            Case "Polynomials"
+                ucrBase.clsRsyntax.AddParameter("new_contrasts", Chr(34) & "contr.poly" & Chr(34))
+            Case "Sum to Zero"
+                ucrBase.clsRsyntax.AddParameter("new_contrasts", Chr(34) & "contr.sum" & Chr(34))
+            Case "User Defined(Overwrite)"
```
Comment (dannyparsons) [inline instat/dlgContrasts.vb]: You don't need user defined case anymore.

**Comment by Lunalo** [comment]
@dannyparsons  

It is now done

**Comment by dannyparsons** [comment]
Are you going to also edit the R method for this as I showed you? Or do you want me to do that?

**Comment by Lunalo** [comment]
I think i will do that

**Pull request merged**